Assets/Scripts/GooglePlayDownloader.cs

```diff
@@ -1,111 +1,35 @@
+    string GetPatchOBBPath();
+    void FetchOBB();
+}
 
-	public static string GetMainOBBPath(string expansionFilePath)
-	{
-		return GetOBBPackagePath(expansionFilePath, "main");
-	}
+public class GooglePlayObbDownloadManager
+{
+    private static AndroidJavaClass m_AndroidOSBuildClass = new AndroidJavaClass("android.os.Build");
+    private static IGooglePlayObbDownloader m_Instance;
 
-	public static string GetPatchOBBPath(string expansionFilePath)
-	{
-		return GetOBBPackagePath(expansionFilePath, "patch");
-	}
+    public static IGooglePlayObbDownloader GetGooglePlayObbDownloader()
+    {
+        if (m_Instance != null)
+            return m_Instance;
 
-	private static string GetOBBPackagePath(string expansionFilePath, string prefix)
-	{
-		if (expansionFilePath == null)
-			return null;
+        if (!IsRunningOnAndroid())
+            return null;
 
-		PopulateOBBData();
-		string filePath = String.Format("{0}/{1}.{2}.{3}.obb", expansionFilePath, prefix, ObbVersion, ObbPackage);
-		return File.Exists(filePath) ? filePath : null;
-	}
+        m_Instance = new GooglePlayObbDownloader();
+        return m_Instance;
+    }
 
-	public static void FetchOBB()
-	{
-		using (AndroidJavaClass unityPlayerClass = new AndroidJavaClass("com.unity3d.player.UnityPlayer"))
-		{
-			AndroidJavaObject currentActivity = unityPlayerClass.GetStatic<AndroidJavaObject>("currentActivity");
-			AndroidJavaObject intent = new AndroidJavaObject("android.content.Intent",
-															currentActivity,
-															new AndroidJavaClass("com.unity3d.plugin.downloader.UnityDownloaderActivity"));
-	
-			const int Intent_FLAG_ACTIVITY_NO_ANIMATION = 0x10000;
-			intent.Call<AndroidJavaObject>("addFlags", Intent_FLAG_ACTIVITY_NO_ANIMATION);
-			intent.Call<AndroidJavaObject>("putExtra", "unityplayer.Activity",
-														currentActivity.Call<AndroidJavaObject>("getClass").Call<string>("getName"));
-			try
-			{
-				currentActivity.Call("startActivity", intent);
-			}
-			catch (Exception e)
-			{
-				Debug.LogError("Exception occurred while attempting to start DownloaderActivity - is the AndroidManifest.xml incorrect?\n" + e.Message);
-			}
-		}
-	}
-	
-	// This code will reuse the package version from the .apk when looking for the .obb
-	// Modify as appropriate
-	private static string ObbPackage;
-	private static int ObbVersion = 0;
-	private static void PopulateOBBData()
-	{
-		if (ObbVersion != 0)
-			return;
-		using (AndroidJavaClass unityPlayerClass = new AndroidJavaClass("com.unity3d.player.UnityPlayer"))
-		{
-			AndroidJavaObject currentActivity = unityPlayerClass.GetStatic<AndroidJavaObject>("currentActivity");
-			ObbPackage = currentActivity.Call<string>("getPackageName");
-			AndroidJavaObject packageInfo = currentActivity.Call<AndroidJavaObject>("getPackageManager").Call<AndroidJavaObject>("getPackageInfo", ObbPackage, 0);
-			ObbVersion = packageInfo.Get<int>("versionCode");
-		}
-	}
+    public static bool IsRunningOnAndroid()
```

Comment:
From the public API point of view, this looks weird in GooglePlayObbDownloadManager. I think renaming it to IsDownloaderAvailable would make more sense.